web/src/lib/components/asset-viewer/detail-panel-star-rating.svelte

```diff
@@ -0,0 +1,98 @@
+<script lang="ts">
+  import { handleError } from '$lib/utils/handle-error';
+  import { updateAsset, type AssetResponseDto } from '@immich/sdk';
+  import { t } from 'svelte-i18n';
+
+  export let asset: AssetResponseDto;
+  export let isOwner: boolean;
+
+  const countStars = 5;
+
+  $: rating = asset.exifInfo?.rating || 0;
+  let currentRating = rating;
+
+  const handleChangeRating = async () => {
+    if (currentRating === asset.exifInfo?.rating) {
+      return;
+    }
+    try {
+      await updateAsset({ id: asset.id, updateAssetDto: { rating: currentRating } });
+    } catch (error) {
+      handleError(error, $t('cant_apply_changes'));
+    }
+    rating = currentRating;
+  };
+</script>
+
+<section class="relative flex px-4">
+  <div class="relative gap-2">
+    <div class="flex">
+      {#each { length: countStars } as _, id}
+        {#if rating > id}
+          <span
+            ><svg
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 24 24"
+              fill="currentColor"
+              transform="scale(0.9,0.9)"
+              class="w-6 h-6 text-yellow-400 cursor-pointer"
```

Comment:
I don't think we need two separate star icons. Why not add styling such as `text-yellow-400` conditionally when `rating > id`?

Reply:
Thank you Michel, just changed it